### DISCUSSION TITLE
Questions about comments/replies, excerpt, profile updates

### DISCUSSION THREAD

**Discussion by belaset** [Automattic/wordpress-activitypub#187] [Q&A]
First, thank you for this plugin! It's really great to be able to share blogposts directly to the fediverse :)

I installed the plugin last night and made a test-post. It works almost perfectly. But then someone left a comment through Mastodon, and I can't respond there because that's not "me" - that is, I don't see how to log on there to reply as my blog-author self rather than with my personal Mastodon account and then, their comment didn't appear on the blog post itself, so I can't respond there either. 

So I guess one question is "is there a way to respond to comments directly on Mastodon" along with "how can I even be notified of comments on Mastodon if I'm not logged in as that user?" and another is "what setting do I need to adjust so that comments show up on the Wordpress blogpost itself?" followed by "if I respond to fediverse-originated comments on the blogpost, will it notify the person who made the comment that they have a reply?"

Another question: if I set the post-sharing to an excerpt, will it still include the featured image from the post?

And I think this is my last question: I have updated my user profile on my Wordpress blog but it doesn't seem to have changed anything when I look at the profile on Mastodon. In fact it has filled in the fields for blog, profile, and website all with my blog's URL. Where can I change those fields, please? 

Thank you again! :)

**Comment by pfefferle** [comment]
Comments should be federated with WordPress. Often times they will be detected as spam, so please check the spam folder for some comments. You can not respond to comments as your blog user on mastodon, but we are working on a feature that lags you comment on your blog and the plugin federates it to mastodon. But this is still work in progress.

**Reply by pfefferle** [reply]
Currently the plugin does not send profile updates, that has to implemented. Good point!

**Reply by pfefferle** [reply]
If you send it as excerpt it should still send the image, but as attachment (at the end of the post) and not inline.

**Reply by belaset** [reply]
> Often times they will be detected as spam, so please check the spam folder for some comments. 

Ah ha! there it is :) now, to make sure WP knows those are *not* spam... thanks again!

**Reply by belaset** [reply]
Okay, I had my test post set to the default of full content, but the picture still ended up after the text, not inline. In the WP post there is a picture between two paragraphs (it is here: https://www.knittingpirate.com/2022/11/18/in-which-the-pirate-test-posts/ ) but when I see it on Mastodon (@Pirate@www.knittingpirate.com) I can see the two paragraphs, then the link to the post, and then the picture.

BTW, really appreciate that the alt-text comes through along with the image :)

**Reply by pfefferle** [reply]
This is sadly a problem of mastodon. It strips all images out of the text.

**Reply by belaset** [reply]
Okay, that makes sense.. so if I write a post with multiple images, will they all be included on Mastodon, or is it only the featured image that is brought over?

**Reply by pfefferle** [reply]
They should all be included, but WordPress has a really buggy way to receive all included images, so I have to rework that a bit. But as I said, the plan is to include all.

**Reply by tfooq** [reply]
> > Often times they will be detected as spam, so please check the spam folder for some comments.
> 
> Ah ha! there it is :) now, to make sure WP knows those are _not_ spam... thanks again!

Thanks to all of you for this. Any tips on how to whitelist ActivityPub comments? I'm using the Akismet anti-spam plugin. I've got everything working, but the spam filter picks them up every time.

**Comment by Mannshoch** [comment]
Can I moderate comments a WordPress receive?
What could be done ,if comments contain problematic content?

**Reply by Mannshoch** [reply]
Is that the Update which it is needed to be able to moderate?
https://github.com/pfefferle/wordpress-activitypub/pull/142

**Reply by pfefferle** [reply]
Hey @Mannshoch it is already like you requested. all comments can be moderated, no comment will be published automatically at the moment!

**Reply by mediaformat** [reply]
Yes, you can unapprove a comment, it will show as pending in the Comments screen, and will not be published on your site
